processing/src/main/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapter.java

```diff
@@ -56,25 +56,29 @@
   private final List<JoinableClause> clauses;
   private final boolean enableFilterPushDown;
 
+  /**
+   * @param baseAdapter A StorageAdapter for the left-hand side base segment
+   * @param clauses The right-hand side clauses. The caller is responsible for ensuring that there are no
+   *                duplicate prefixes or prefixes that shadow each other across the clauses
+   * @param enableFilterPushDown Whether to enable filter push down optimizations to the base segment
+   */
   HashJoinSegmentStorageAdapter(
       StorageAdapter baseAdapter,
-      List<JoinableClause> clauses
+      List<JoinableClause> clauses,
+      final boolean enableFilterPushDown
   )
   {
     this.baseAdapter = baseAdapter;
     this.clauses = clauses;
-    this.enableFilterPushDown = QueryContexts.DEFAULT_ENABLE_JOIN_FILTER_PUSH_DOWN;
+    this.enableFilterPushDown = enableFilterPushDown;
   }
 
   HashJoinSegmentStorageAdapter(
```

Comment:
`@VisibleForTesting` Looks like this is only used in tests now

Reply:
Added annotation